File: Practica 5/python/InterfazGrafica.py
import tkinter as tk
from Persona import Persona
class InterfazGrafica(tk.Tk):
    def __init__(self):
        super().__init__()
        self.title("Formulario de Información")
        self.geometry("620x400")
        self.create_widgets()
        self.persona = Persona("a", "b", "c", "d", "e", "f", "g")
        
    def create_widgets(self):
        lblNombre = tk.Label(self, text="Nombre:")
        lblFechaNacimiento = tk.Label(self, text="Fecha de Nacimiento (DD/MM/AAAA):")
        lblOcupacion = tk.Label(self, text="Ocupación:")

        self.txtNombre = tk.Entry(self)
        self.txtApellidoPaterno = tk.Entry(self)
        self.txtApellidoMaterno = tk.Entry(self)
        self.txtDiaNacimiento = tk.Entry(self)
        self.txtMesNacimiento = tk.Entry(self)
        self.txtAnioNacimiento = tk.Entry(self)
        self.txtOcupacion = tk.Entry(self)

        self.txtAreaInformacion = tk.Text(self, height=10, width=40)
        self.txtAreaInformacion.config(state=tk.DISABLED)
        
        # Botones con funciones
        btnSetNombre = tk.Button(self, text="Actualizar Nombre", command=self.actualizar_nombre)
        btnSetNacimiento = tk.Button(self, text="Actualizar Nacimiento", command=self.actualizar_nacimiento)
        btnSetOcupacion = tk.Button(self, text="Actualizar Ocupación", command=self.actualizar_ocupacion)
        btnMostrar = tk.Button(self, text="Mostrar Información", command=self.mostrar_informacion)
        btnOcultar = tk.Button(self, text="Ocultar Información", command=self.ocultar_informacion)
        btnActualizarGlobal = tk.Button(self, text="Actualizar Global", command=self.actualizar_informacion_global)

        #Interfaz en pantalla
        lblNombre.grid(row=0, column=0)
        self.txtNombre.grid(row=0, column=1)
        tk.Label(self, text="Apellido Paterno:").grid(row=1, column=0)
        self.txtApellidoPaterno.grid(row=1, column=1)
        tk.Label(self, text="Apellido Materno:").grid(row=2, column=0)
        self.txtApellidoMaterno.grid(row=2, column=1)
        lblFechaNacimiento.grid(row=3, column=0)
        self.txtDiaNacimiento.grid(row=3, column=1)
        tk.Label(self, text="/").grid(row=3, column=2)
        self.txtMesNacimiento.grid(row=3, column=3)
        tk.Label(self, text="/").grid(row=3, column=4)
        self.txtAnioNacimiento.grid(row=3, column=5)
        lblOcupacion.grid(row=4, column=0)
        self.txtOcupacion.grid(row=4, column=1)

        self.txtAreaInformacion.grid(row=5, column=0, columnspan=2)
        
        btnSetNombre.grid(row=6, column=0)
        btnSetNacimiento.grid(row=6, column=1)
        btnSetOcupacion.grid(row=7, column=0)
        btnMostrar.grid(row=7, column=1)
        btnOcultar.grid(row=8, column=0)
        btnActualizarGlobal.grid(row=8, column=1)
        
    
    # Funciones de los botones
    def actualizar_nombre(self):
        nombre = self.txtNombre.get()
        apellido_paterno = self.txtApellidoPaterno.get()
        apellido_materno = self.txtApellidoMaterno.get()
        
        self.persona.set_nombre(nombre, apellido_paterno, apellido_materno)
        self.persona.calculate_id()
        
        self.txtAreaInformacion.config(state=tk.NORMAL)
        self.txtAreaInformacion.delete("1.0", tk.END)
        self.txtAreaInformacion.insert(tk.END, self.persona.__str__())
        self.txtAreaInformacion.config(state=tk.DISABLED)
        
    def actualizar_nacimiento(self):
        dia_nacimiento = self.txtDiaNacimiento.get()
        mes_nacimiento = self.txtMesNacimiento.get()
        anio_nacimiento = self.txtAnioNacimiento.get()

        self.persona.set_nacimiento(dia_nacimiento, mes_nacimiento, anio_nacimiento)
        self.persona.calculate_id()
        self.persona.calculate_edad()

        self.txtAreaInformacion.config(state=tk.NORMAL)
        self.txtAreaInformacion.delete("1.0", tk.END)
        self.txtAreaInformacion.insert(tk.END, self.persona.__str__())
        self.txtAreaInformacion.config(state=tk.DISABLED)
        
    def actualizar_ocupacion(self):
        ocupacion = self.txtOcupacion.get()

        self.persona.set_ocupacion(ocupacion)

        self.txtAreaInformacion.config(state=tk.NORMAL)
        self.txtAreaInformacion.delete("1.0", tk.END)
        self.txtAreaInformacion.insert(tk.END, self.persona.__str__())
        self.txtAreaInformacion.config(state=tk.DISABLED)
    
    def mostrar_informacion(self):
        self.txtAreaInformacion.config(state=tk.NORMAL)
        self.txtAreaInformacion.delete("1.0", tk.END)
        self.txtAreaInformacion.insert(tk.END, "Aquí se mostrará la información.")
        self.txtAreaInformacion.config(state=tk.DISABLED)
    def ocultar_informacion(self):
        self.txtAreaInformacion.config(state=tk.NORMAL)
        self.txtAreaInformacion.delete("1.0", tk.END)
        self.txtAreaInformacion.config(state=tk.DISABLED)
    def actualizar_informacion_global(self):
        nombre = self.txtNombre.get()
        apellido_paterno = self.txtApellidoPaterno.get()
        apellido_materno = self.txtApellidoMaterno.get()
        dia_nacimiento = self.txtDiaNacimiento.get()
        mes_nacimiento = self.txtMesNacimiento.get()
        anio_nacimiento = self.txtAnioNacimiento.get()
        ocupacion = self.txtOcupacion.get()

        self.persona.set_nombre(nombre, apellido_paterno, apellido_materno)
        self.persona.set_nacimiento(dia_nacimiento, mes_nacimiento, anio_nacimiento)
        self.persona.set_ocupacion(ocupacion)

        self.persona.calculate_edad()
        self.persona.calculate_id()

        self.txtAreaInformacion.config(state=tk.NORMAL)
        self.txtAreaInformacion.delete("1.0", tk.END)
        self.txtAreaInformacion.insert(tk.END, self.persona.__str__())
        self.txtAreaInformacion.config(state=tk.DISABLED)

if __name__ == "__main__":
    app = InterfazGrafica()
    app.mainloop()

File: Practica 5/java/InterfazGrafica.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class InterfazGrafica extends JFrame {
    private JTextField txtNombre;
    private JTextField txtApellidoPaterno;
    private JTextField txtApellidoMaterno;
    private JTextField txtDiaNacimiento;
    private JTextField txtMesNacimiento;
    private JTextField txtAnioNacimiento;
    private JTextField txtOcupacion;
    private JTextArea txtAreaInformacion;
    private JButton btnSetNombre;
    private JButton btnSetNacimiento;
    private JButton btnSetOcupacion;
    private JButton btnMostrar;
    private JButton btnOcultar;
    private JButton btnActualizarGlobal;

    Persona p1 = new Persona();
    
    public InterfazGrafica() {
        // Configuración de la ventana principal
        setTitle("Formulario de Información");
        setSize(450, 400);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setLocationRelativeTo(null);

        // Creación de los componentes
        JLabel lblNombre = new JLabel("Nombre:");
        JLabel lblFechaNacimiento = new JLabel("Fecha de Nacimiento (DD/MM/AAAA):");
        JLabel lblOcupacion = new JLabel("Ocupación:");

        txtNombre = new JTextField(20);
        txtApellidoPaterno = new JTextField(10);
        txtApellidoMaterno = new JTextField(10);
        txtDiaNacimiento = new JTextField(2);
        txtMesNacimiento = new JTextField(2);
        txtAnioNacimiento = new JTextField(4);
        txtOcupacion = new JTextField(20);

        btnSetNombre = new JButton("Actualizar Nombre");
        btnSetNacimiento = new JButton("Actualizar Nacimiento");
        btnSetOcupacion = new JButton("Actualizar Ocupación");
        btnMostrar = new JButton("Mostrar Información");
        btnOcultar = new JButton("Ocultar Información");
        btnActualizarGlobal = new JButton("Actualizar Global");

        txtAreaInformacion = new JTextArea(10, 20);
        txtAreaInformacion.setEditable(false);

        // Creación del panel para organizar los componentes
        JPanel panel = new JPanel();
        panel.setLayout(new GridLayout(8, 2));
        panel.add(lblNombre);
        panel.add(txtNombre);
        panel.add(new JLabel("Apellido Paterno:"));
        panel.add(txtApellidoPaterno);
        panel.add(new JLabel("Apellido Materno:"));
        panel.add(txtApellidoMaterno);
        panel.add(lblFechaNacimiento);
        JPanel panelFecha = new JPanel();
        panelFecha.add(txtDiaNacimiento);
        panelFecha.add(new JLabel("/"));
        panelFecha.add(txtMesNacimiento);
        panelFecha.add(new JLabel("/"));
        panelFecha.add(txtAnioNacimiento);
        panel.add(panelFecha);
        panel.add(lblOcupacion);
        panel.add(txtOcupacion);
        panel.add(btnSetNacimiento);
        panel.add(btnSetNombre);
        panel.add(btnSetOcupacion);
        panel.add(btnActualizarGlobal);
        panel.add(btnOcultar);
        panel.add(btnMostrar);

        // Agregar el panel a la ventana
        add(panel, BorderLayout.NORTH);
        add(new JScrollPane(txtAreaInformacion), BorderLayout.CENTER);

        // Agregar eventos a los botones
        
        btnSetNombre.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                actualizarNombre();
            }
        });
        
        btnSetNacimiento.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                actualizarNacimiento();
            }
        });
        
        btnSetOcupacion.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                actualizarOcupacion();
            }
        });
        
        btnMostrar.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                mostrarInformacion();
            }
        });

        btnOcultar.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                ocultarInformacion();
            }
        });

        btnActualizarGlobal.addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                actualizarInformacionGlobal();
            }
        });
    }
    
    private void actualizarCalculos(){
        p1.calculateEdad();
        p1.calculateID();
    }
    
    private void actualizarNombre(){
        String nombre = txtNombre.getText();
        String apellidoPaterno = txtApellidoPaterno.getText();
        String apellidoMaterno = txtApellidoMaterno.getText();
        
        if (nombre.isEmpty() || apellidoPaterno.isEmpty() || apellidoMaterno.isEmpty() ) {
            JOptionPane.showMessageDialog(this, "Por favor, complete todos los campos.", "Error", JOptionPane.ERROR_MESSAGE);
            return;
        }
        
        p1.setNombre(nombre, apellidoPaterno, apellidoMaterno);
        actualizarCalculos();
        
        // Mostrar el mensaje en el área de texto
        txtAreaInformacion.setText(p1.toString());
    }
    
    private void actualizarNacimiento(){
        String dia = txtDiaNacimiento.getText();
        String mes = txtMesNacimiento.getText();
        String anio = txtAnioNacimiento.getText();

        if (dia.isEmpty() || mes.isEmpty() || anio.isEmpty()) {
            JOptionPane.showMessageDialog(this, "Por favor, complete todos los campos.", "Error", JOptionPane.ERROR_MESSAGE);
            return;
        }

        p1.setNacimiento(dia, mes, anio );
        actualizarCalculos();
        
        // Mostrar el mensaje en el área de texto
        txtAreaInformacion.setText(p1.toString());
    }
    
    private void actualizarOcupacion(){
        String ocupacion = txtOcupacion.getText();

        if (ocupacion.isEmpty()) {
            JOptionPane.showMessageDialog(this, "Por favor, complete todos los campos.", "Error", JOptionPane.ERROR_MESSAGE);
            return;
        }

        p1.setOcupacion(ocupacion);
        actualizarCalculos();
        
        // Mostrar el mensaje en el área de texto
        txtAreaInformacion.setText(p1.toString());
    }
    
    private void actualizarInformacionGlobal() {
        // Obtener los valores de los campos
        String nombre = txtNombre.getText();
        String apellidoPaterno = txtApellidoPaterno.getText();
        String apellidoMaterno = txtApellidoMaterno.getText();
        String dia = txtDiaNacimiento.getText();
        String mes = txtMesNacimiento.getText();
        String anio = txtAnioNacimiento.getText();
        String ocupacion = txtOcupacion.getText();

        p1 = new Persona(nombre, apellidoPaterno, apellidoMaterno, ocupacion, dia, mes, anio);
        actualizarCalculos();
        
        // Validar que no haya campos vacíos
        if (nombre.isEmpty() || apellidoPaterno.isEmpty() || apellidoMaterno.isEmpty() ||
            dia.isEmpty() || mes.isEmpty() || anio.isEmpty() || ocupacion.isEmpty()) {
            JOptionPane.showMessageDialog(this, "Por favor, complete todos los campos.", "Error", JOptionPane.ERROR_MESSAGE);
            return;
        }

        // Mostrar el mensaje en el área de texto
        txtAreaInformacion.setText(p1.toString());
    }

    private void mostrarInformacion() {
        txtAreaInformacion.setVisible(true);
    }

    private void ocultarInformacion() {
        txtAreaInformacion.setVisible(false);
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                new InterfazGrafica().setVisible(true);
            }
        });
    }
}
